server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -328,13 +306,13 @@ public void parse(ParseContext context) throws IOException {
             }
 
             List<IndexableField> fields = new ArrayList<>();
-            if (fieldType.indexOptions() != IndexOptions.NONE || fieldType.hasDocValues()) {
+            if (mappedFieldType.isSearchable() || mappedFieldType.hasDocValues()) {
```

Comment:
what happens with these checks once isSearchable returns true yet there is no index? Do we need to distinguish between the two?

Reply:
We will need to distinguish between them in the future, yes.  This is something of a shim though - the plan is to eventually move everything to parametrized mappers (see #58663) which will change how this works again.